python/app/input_widget.py

```diff
@@ -184,7 +181,7 @@ def execute(self):
 
         # exec
         else:
-            with nested(self._stdout_redirect, self._stdin_redirect):
```

Comment:
I can't test on Python 2, but I'm concerned this might break it?

Reply:
Why did you remove the nested call? I thought this was introduced in Python 2.6 because you can't have multiple values with a single `with` statement IIRC?

Reply:
https://stackoverflow.com/questions/893333/multiple-variables-in-a-with-statement
Python 3 doesn't support `nested`, instead the approach is to just provide a tuple.

Reply:
Could you nest the calls with two `with` then, so as to preserve 2.6 compatibility? I'm sure Francis will find a way to test it.

Reply:
good idea!